extensions/bitwarden/src/search.tsx

```diff
@@ -0,0 +1,191 @@
+import {
+  ActionPanel,
+  List,
+  Icon,
+  showToast,
+  ToastStyle,
+  PushAction,
+  Detail,
+  CopyToClipboardAction,
+  getPreferenceValues,
+} from "@raycast/api";
+import { Item, Folder } from "./types";
+import { useEffect, useState } from "react";
+import treeify from "treeify";
+import { filterNullishPropertiesFromObject, codeBlock, titleCase } from "./utils";
+import { useBitwarden as useSessionToken } from "./hooks";
+import { TroubleshootingGuide, UnlockForm } from "./components";
+import { existsSync } from "fs";
+import { BitwardenApi } from "./api";
+import { dirname } from "path";
+
+const { cliPath, clientId, clientSecret, fetchFavicons } = getPreferenceValues();
+process.env.PATH = dirname(cliPath)
+const bitwardenApi = new BitwardenApi(clientId, clientSecret);
+
+export default function ListCommand(): JSX.Element {
+  if (!existsSync(cliPath)) {
+    return <TroubleshootingGuide />;
+  }
+
+  const [sessionToken, setSessionToken] = useSessionToken(bitwardenApi);
+
+  if (sessionToken === null) return <UnlockForm setSessionToken={setSessionToken} bitwardenApi={bitwardenApi} />;
+
+  return <ItemList bitwardenApi={bitwardenApi} sessionToken={sessionToken}/>;
+}
+
+function ItemList(props: { bitwardenApi: BitwardenApi; sessionToken: string | undefined }) {
+  const { bitwardenApi, sessionToken } = props;
+  const [state, setState] = useState<{ folders: Folder[]; items: Item[] }>();
+
+  const folderMap: Record<string, Folder> = {};
+  for (const folder of state?.folders || []) {
+    if (folder.id) folderMap[folder.id] = folder;
+  }
+
+  async function loadItems(sessionToken: string) {
+    try {
+      const [items, folders] = await Promise.all([
+        bitwardenApi.listItems<Item>("items", sessionToken),
+        bitwardenApi.listItems<Folder>("folders", sessionToken),
+      ]);
+
+      setState({ items, folders });
+    } catch (error) {
+      showToast(ToastStyle.Failure, "Failed to search vault", "Vault is locked");
```

Comment:
It's unclear to me if a locked vault is the _only_ reason this could throw an error? It would improve the user experience if we used the `error` value in the toast, but I'm not sure what information those `error` values might contain. Thoughts?